### PR TITLE
fix duplicate domain error

### PR DESCRIPTION
**Pull request by maxchehab**
Uses an existing organization if one exists.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -40,15 +40,30 @@
...
-    name: domain.partition('.').first,
+  organizations = WorkOS::Portal.list_organizations(
+   domains: [domain],
   )
 
-  portal_link = WorkOS::Portal.generate_link(
-    intent: 'sso',
-    organization: organization.id
-  )
+  if organizations.data.length > 0
```
Comment (maxchehab) [inline app.rb]: is this the most idiomatic way to check for an non-empty array?

Comment (willmanduffy) [reply app.rb:47]: Yeah probably without lodash

Comment (willmanduffy) [reply app.rb]: empty? should do the trick

```rb
irb(main):001:0> [].empty?
=> true
```

Comment (maxchehab) [reply app.rb]: beautiful

File: app.rb [hunk 2]
```diff
@@ -40,15 +40,30 @@
...
+     domains: [domain],
+     name: domain.partition('.').first,
+    )
+
+    portal_link = WorkOS::Portal.generate_link(
+     intent: 'sso',
+     organization: organization.id
+    )
+
+    redirect portal_link
```
Comment (willmanduffy) [inline app.rb:58]: If this happens regardless of the branch we could have this redirect outside of it

```rb
if organizations.data.empty?
  ...
else
  ...
end

redirect portal_link
```

local variables can escape the scope of the block they are defined in (within the closure of the method). You can also assign the return value of a block to a variable

```rb
organization = if organizations.data.empty?
  ...
else
 ...
end

portal_link = WorkOS::Portal.generate_link(
     intent: 'sso',
     organization: organization.id
 )

redirect portal_link
```

**Review by willmanduffy — APPROVED** [review]

**Pull request merged**